test/index.ts

```diff
@@ -404,6 +404,27 @@ describe('.getToken()', () => {
     assert(creds.privateKey);
     assert(creds.clientEmail);
   });
+
+  // see: https://github.com/googleapis/google-api-nodejs-client/issues/1614
+  it('should throw exception if readFile not available, and keyFile provided',
+     async () => {
+       // Fake an environment in which fs.readFile does not
+       // exist. This is the same as when running in the browser.
+       delete require.cache[require.resolve('../src')];
+       delete require.cache[require.resolve('fs')];
+       const fs = require('fs');
+       delete fs.readFile;
+       const {GoogleToken} = require('../src');
+
+       let message;
+       try {
+         const gtoken = new GoogleToken(TESTDATA_KEYFILEJSON);
```

Comment:
note `gtoken.getCredentials` is async, `assert.throw` does not seem to work with async functions?

Reply:
That is very correct.  For that you need to use `assert.rejects`.